src/goto-symex/symex_decl.cpp

```diff
@@ -69,7 +69,11 @@ void goto_symext::symex_decl(statet &state, const symbol_exprt &expr)
   symex_renaming_levelt::increase_counter(level2_it);
   const bool record_events=state.record_events;
   state.record_events=false;
-  state.rename<goto_symex_statet::L2>(ssa, ns);
+  exprt expr_l2 = state.rename<statet::L2>(std::move(ssa), ns);
+  INVARIANT(
+    expr_l2.id() == ID_symbol && expr_l2.get_bool(ID_C_SSA_symbol),
+    "symbol to declare should not be replaced by constant propagation");
```

Comment:
Fair enough, but unrelated to the other changes here?

Reply:
We need these conditions to be sure `to_ssa_expr` will not failed in the next line. Before `ssa` was modified in-place so there was no conversion here.

Reply:
Ah, indeed that possible problem was hidden before. But `to_ssa_expr` should already do the same checking for you?